zkevm-circuits/src/evm_circuit/util.rs

```diff
@@ -499,10 +499,162 @@ impl<F: FieldExt, const N: usize> Expr<F> for RandomLinearCombination<F, N> {
     }
 }
 
-pub(crate) type Word<F> = RandomLinearCombination<F, 32>;
+#[derive(Clone, Debug)]
+pub(crate) struct WordLimbs<T, const N: usize> {
+    pub limbs: [T; N],
+}
+
+pub(crate) type Word2<T> = WordLimbs<T, 2>;
+
+pub(crate) type Word4<T> = WordLimbs<T, 4>;
+
+pub(crate) type Word16<T> = WordLimbs<T, 16>;
+
+pub(crate) type Word32<T> = WordLimbs<T, 32>;
+
+impl<T, const N: usize> WordLimbs<T, N> {
+    fn new(limbs: [T; N]) -> Self {
+        Self { limbs }
+    }
+
+    fn n() -> usize {
+        N
+    }
+}
+
+impl<T: Default, const N: usize> Default for WordLimbs<T, N> {
+    fn default() -> Self {
+        Self {
+            limbs: [(); N].map(|_| T::default()),
+        }
+    }
+}
+
+pub(crate) trait WordCells<F: FieldExt, const N: usize> {
+    fn assign<const N1: usize>(
+        &self,
+        region: &mut CachedRegion<'_, '_, F>,
+        offset: usize,
+        bytes: Option<[u8; N1]>,
+    ) -> Result<Vec<AssignedCell<F, F>>, Error>;
+
+    fn expr(&self) -> WordLimbs<Expression<F>, N>;
+}
+
+impl<F: FieldExt, const N: usize> WordCells<F, N> for WordLimbs<Cell<F>, N> {
+    fn assign<const N1: usize>(
+        &self,
+        region: &mut CachedRegion<'_, '_, F>,
+        offset: usize,
+        bytes: Option<[u8; N1]>,
+    ) -> Result<Vec<AssignedCell<F, F>>, Error> {
+        assert_eq!(N1 % N, 0); // assure N|N1
```

Comment:
Maybe we can use https://docs.rs/static_assertions/latest/static_assertions/ for these assertions that should be performed at compile time?

Reply:
Yes I plan to use this. However, `static_assertion` this library do not work on const generic, see issue here
https://github.com/nvzqz/static-assertions-rs/issues/40
I will try the workaround later.